service/awsconfig/v7/resource/lifecycle/resource.go

```diff
@@ -14,29 +15,32 @@ const (
 type ResourceConfig struct {
 	Clients awsclient.Clients
 	Logger  micrologger.Logger
+	Service *cloudformationservice.CloudFormation
 }
 
 type Resource struct {
 	clients awsclient.Clients
 	logger  micrologger.Logger
+	service *cloudformationservice.CloudFormation
 }
 
 func NewResource(config ResourceConfig) (*Resource, error) {
-	if config.Clients.CloudFormation == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.Clients.CloudFormation must not be empty", config)
-	}
-	if config.Clients.EC2 == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.Clients.EC2 must not be empty", config)
+	if config.Clients.AutoScaling == nil {
```

Comment:
In my discovery PR I noticed I need the autoscaling client, not the others. For the cloudformation magic there is the new service.